### PR TITLE
Favorite Places Endpoint

### LINKED ISSUE #39: Creating, reading and deleting favorite places
## Issue Description
Implement creating, reading and deleting favorite user locations. The update method will not be necessary.
For reading, only the get by user endpoint is required.
The necessary endpoints must be protected with authentication and it should not be possible for a user to tamper with another user's favorite locations.

## Model of favorite places
The relationship between the favorite place table and user table is 1 to N, that is, a user can have N favorite places.
The model attributes will be:
* string user, the foreign key with the username of the user
* integer id_place, foreign with the id of place
* float latitude
* float longitude

All the fields are required.

## Tasks:
- [x] Implement the model
- [x] Implement post endpont
- [x] Implement delete endpoint
- [x] Implement get by user endpoint
- [x] Protect the endpoints
- [x] Make tests
- [x] Update swagger

### PR DESCRIPTION
**Pull request by herickport**
## Issue 
Resolves #39 

## Description

Implemented create, read and delete favorite user locations. 
The endpoints are protected with authentication and it is not possible for a user to change another user's favorite locations.
Updated swagger documentation accordingly.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

## How to validate?

Check in route 'api/places/' if get, post and delete methods are working correctly.
Run the tests.

### CODE REVIEW

File: src/views/favorite_places.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@favorite_places_blueprint.route('/places/<int:id_place>', methods=['DELETE'])
+@validate_header
+@validate_token
+def delete_favorite_place_by_id(username, id_place):
+    response, status = controller.delete_favorite_place(username, id_place)
+
+    return create_response(response, status)
+
+@favorite_places_blueprint.route('/places/', methods=['GET'])
+@validate_header
```
Comment (silvasara) [inline src/views/favorite_places.py:31]: This decorator is only necessary in POST, PATCH or PUT methods

File: src/views/favorite_places.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@validate_token
+def post_favorite_place(username):
+    response, status = controller.create_favorite_place(
+        request.json, username)
+
+    return create_response(response, status)
+
+@favorite_places_blueprint.route('/places/<int:id_place>', methods=['DELETE'])
+@validate_header
```
Comment (silvasara) [inline src/views/favorite_places.py:23]: This decorator is only necessary in POST, PATCH or PUT methods

File: src/views/favorite_places.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+favorite_places_blueprint = Blueprint('favorite_places', __name__, url_prefix='/api')
+CORS(favorite_places_blueprint)
+
+
+@favorite_places_blueprint.route('/places/', methods=['POST'])
+@validate_header
+@validate_token
+def post_favorite_place(username):
+    response, status = controller.create_favorite_place(
+        request.json, username)
```
Comment (silvasara) [inline src/views/favorite_places.py:18]: In the function `controller.create_favorire_place`the order of the parameters is defined differently from what is being passed in the function call:
https://github.com/fga-eps-mds/2020.1-stay-safe-user-service/blob/857fbbf086e303de481d55fe2cfd95cc3001307d/src/controllers/favorite_places.py#L16

So when I try to post, I have this error:
image

**Review by silvasara — CHANGES_REQUESTED** [review]

**Comment by lboaventura25** [comment]
Thanks, @silvasara. We fix the problems!

**Review by silvasara — APPROVED** [review]

**Review by renan601 — APPROVED** [review]

**Pull request merged**